File: src/main/java/io/junseok/mealmate/domain/membercategory/entity/MemberCategory.java
package io.junseok.mealmate.domain.membercategory.entity;

import io.junseok.mealmate.domain.member.entity.Member;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "member_category")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class MemberCategory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_category_id")
    private Long memberCategoryId;

    @Column(name = "category_name")
    private String categoryName;

    @JoinColumn(name = "member_id")
    @ManyToOne(fetch = FetchType.LAZY)
    private Member member;

    public void modifyCategoryName(String categoryName){
        this.categoryName=categoryName;
    }
}

File: src/main/java/io/junseok/mealmate/domain/membercategory/entity/MemberCategory.kt
package io.junseok.mealmate.domain.membercategory.entity

import io.junseok.mealmate.domain.member.entity.Member
import javax.persistence.*

@Entity
@Table(name = "member_category")
class MemberCategory(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_category_id")
    var memberCategoryId: Long? = null,

    @Column(name = "category_name")
    var categoryName: String,

    @JoinColumn(name = "member_id")
    @ManyToOne(fetch = FetchType.LAZY)
    val member: Member
) {
    fun modifyCategoryName(categoryName: String) {
        this.categoryName = categoryName
    }
}
